File: programming/leetcode/k_closest_point_to_origin.cpp
// https://leetcode.com/problems/k-closest-points-to-origin/submissions/
// Runtime: 388 ms, faster than 44.33% of C++ online submissions
// Memory Usage: 34.4 MB, less than 100.00% of C++ online submissions
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int K) {
        std::sort(points.begin(), points.end(),
                 [](const std::vector<int> &a, const std::vector<int> &b){
                     return (a[0] * a[0] + a[1] * a[1]) < 
                         (b[0] * b[0] + b[1] * b[1]);
                 });
        return std::vector< std::vector<int> >(points.begin(), points.begin() + K);
    }
};

File: programming/leetcode/k_closest_point_to_origin.py
# https://leetcode.com/problems/k-closest-points-to-origin/
# Runtime: 664 ms, faster than 93.00% of Python3 online submissions 
# Memory Usage: 19.1 MB, less than 5.80% of Python3 online submissions
class Solution:
    def kClosest(self, points: List[List[int]], K: int) -> List[List[int]]:
        points.sort(key=lambda point: point[0] ** 2 + point[1] ** 2)
        return points[:K]
        